### DISCUSSION TITLE
Enter a IP address

### DISCUSSION THREAD

**Discussion by workshop748** [ThusharaX/C-programming#107] [Q&A] [answered]
Hey I'm a fairly new programming in c++. I was just curious if there a way to enter a ip address in c++ to help calculate a subnet

**Comment by palonza** [comment, marked as answer]
hi @workshop748,

in this example, 'calculateSubnetMask' takes prefix length as input and generates the corresponding subnet mask using bitwise operations and main function prompts the user to enter IP and prefix... is basic but is working... please let me know if work to you... regards

```cpp
#include <iostream>
#include <string>
#include <sstream>
#include <bitset>

// Function to calculate the subnet mask based on the prefix length
std::string calculateSubnetMask(int prefixLength) {
    std::bitset<32> bits;
    for (int i = 0; i < prefixLength; ++i) {
        bits.set(i, true);
    }
    std::stringstream ss;
    ss << (bits.to_ulong() >> 24) << '.' << ((bits.to_ulong() >> 16) & 255) << '.' 
       << ((bits.to_ulong() >> 8) & 255) << '.' << (bits.to_ulong() & 255);
    return ss.str();
}

int main() {
    std::string ipAddress;
    int prefixLength;

    // Read IP address and prefix length from the user
    std::cout << "Enter IP address (e.g., 192.168.0.1): ";
    std::cin >> ipAddress;
    std::cout << "Enter prefix length (e.g., 24): ";
    std::cin >> prefixLength;

    // Calculate subnet mask
    std::string subnetMask = calculateSubnetMask(prefixLength);

    // Print the calculated subnet mask
    std::cout << "Subnet Mask: " << subnetMask << std::endl;

    return 0;
}

```

**Reply by workshop748** [reply]
It does thanks

**Reply by palonza** [reply]
I'm very happy to hear it. I would appreciate it if you **could mark your question as resolved**, so we can close the case and help me.

**Reply by workshop748** [reply]
How do I do that sorry this is still very new to me.

**Reply by palonza** [reply]
Hi @workshop748 , in the thread of the conversation, above my chat, you should find a button that says something like this: "mark as answered", click there. Thanks in advance!
